### PR TITLE
[Fix]: Heatmap graph click issue

### PR DESCRIPTION
**Pull request by nxanil**
@natabal 

Fixed issue #403. Please have a look.

### CODE REVIEW

**Comment by natabal** [comment]
@nxanil This does not fix the issue.

**Comment by nxanil** [comment]
@natabal 

The above issue and #405 issue have been fixed. Please have a look.

File: Graphs/HeatmapGraph/index.js [hunk 1]
```diff
@@ -695,21 +700,28 @@ const HeatmapGraph = (props) => {
     if(getLeftMargin() !== margin.left && getLeftMargin() !== 0 && showGraph === false) {
         setShowGraph(true);
     } 
-    
+
```
Comment (natabal) [inline Graphs/HeatmapGraph/index.js:703]: This Tooltip should be moved outside of this component.

**Comment by natabal** [comment]
@nxanil Two issues noticed:
- Graph is taking a long time to render (minutes). The ES search is quick and there are no errors in the console. 
- When clicking the next interval ('>') the graph renders incorrectly

heatmap

**Comment by chaitanyahalbe** [comment]
@nxanil @natabal Saw this issue when I used this PR in my FEC PRs. It's not happening with the code in vis-graphs 0.0.
image
image
`

**Comment by chaitanyahalbe** [comment]
@nxanil Do you have an estimate for this fix?

**Comment by nxanil** [comment]
@chaitanyahalbe 

I have fixed the above issue in this PR https://github.com/nuagenetworks/vsd-react-ui/pull/4517, Please verify once.

Thanks!

**Comment by chaitanyahalbe** [comment]
> @chaitanyahalbe
> 
> I have fixed the above issue in this PR [nuagenetworks/vsd-react-ui#4517](https://github.com/nuagenetworks/vsd-react-ui/pull/4517), Please verify once.
> 
> Thanks!

@nxanil Thanks! This has resolved the above issue I had reported.

**Comment by chaitanyahalbe** [comment]
@nxanil Although I still see an issue which @natabal reported in case of going backward and forward in time with `<` and `>`. 
Screenshots:
image
image

**Comment by natabal** [comment]
@nxanil There are still issues here: 
- once you click on one of the '<', '>' all the graphs are actually reloaded
- not all the graphs are populated properly even though there is data.

heatmap-loading

**Comment by chaitanyahalbe** [comment]
@nxanil @natabal The issue with going backward and forward in time is fixed. But observed the below issue now.

The heatmap graph work fine for any **Time interval** filter value except for **Custom Time** value. The heatmap graph is not reloaded upon selecting a custom time interval. See the GIF below.
Sep-22-2020 11-19-14

**Comment by natabal** [comment]
@chaitanyahalbe This PR must be verified with multiple heatmaps on the same page as well. Please verify it with your FEC visualizations PR.

**Comment by chaitanyahalbe** [comment]
> @chaitanyahalbe This PR must be verified with multiple heatmaps on the same page as well. Please verify it with your FEC visualizations PR.

Tested this on FEC, IKE and NATT dashboards. Found above issues.

**Comment by nxanil** [comment]
@natabal  @chaitanyahalbe 

Above issues has been fixed into PR https://github.com/nuagenetworks/vsd-react-ui/pull/4549. Please verify once.

**Comment by chaitanyahalbe** [comment]
@nxanil @natabal All the issues reported in this PR and nuagenetworks/vsd-react-ui#4549 are fixed. Working as expected. Thanks!

**Review by natabal — APPROVED** [review]

**Pull request merged**